orchestrator/core/services/plugin_upload_service.py

```
@@ -0,0 +1,219 @@
+        from core.models.marketplace_plugins import (
+            MarketplacePlugin,
+            PluginSyncHistory,
+        )
+
+        # ------------------------------------------------------------------
+        # 1. Extract zip and validate manifest
+        # ------------------------------------------------------------------
+        try:
+            zf = zipfile.ZipFile(io.BytesIO(zip_bytes))
+        except zipfile.BadZipFile:
+            raise ValueError("Uploaded file is not a valid zip archive")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)